File: apA_analyticSolutions/matlab/fixed_critical_1d1g.m
clear
close all

Lx = 1;
LW = 2;
ntest = 100;
num_dots = 11;
dot_size = 50;

diffusion = 1;
qfixed = 1;
sigma_r = 1;
nusigma_f = 2;

Bf = sqrt(sigma_r/diffusion);

xtest = linspace(0,Lx,ntest);

fixed = @(x)((qfixed/Bf^2)*(1-cosh(Bf*x))+ ...
    ((qfixed/Bf^2)*(cosh(Bf*Lx)-1)/(sinh(Bf*Lx)))*sinh(Bf*x));
critical = @(x)(sin((pi/Lx)*x));

ytestf = zeros(size(xtest));
ytestc = zeros(size(xtest));
for i = 1:length(xtest)
    ytestf(i) = fixed(xtest(i));
    ytestc(i) = critical(xtest(i));
end
ytestc = ytestc * norm(ytestf,inf);

figure
hold on
title('One-Dimension, One-Group, Fixed and Critical')
plot(xtest,ytestc,'-','LineWidth',LW)
plot(xtest,ytestf,'--','LineWidth',LW)
xlabel('x [cm]')
ylabel('\phi(x)')
legend({'Criticality', 'Fixed Source'})
hold off

set(gca,'FontName','Times New Roman','FontSize',12);
print('../figs/fixed_critical.eps','-depsc2');
close(gcf)

File: apA_analyticSolutions/python/fixed_critical_1d1g.py
import numpy as np
import matplotlib.pyplot as plt
import numpy.linalg

LW = 2
FN = "Times New Roman"
FS = 12
plt.rc("lines", lw=LW)
plt.rc("mathtext", fontset="stix")
plt.rc("font", family=FN, size=FS)

Lx = 1.0
ntest = 100
num_dots = 11
dot_size = 50

diffusion = 1.0
qfixed = 1.0
sigma_r = 1.0
nusigma_f = 2.0

Bf = np.sqrt(sigma_r / diffusion)

xtest = np.linspace(0.0, Lx, ntest)

fixed = lambda x: (
    (qfixed / Bf**2) * (1 - np.cosh(Bf * x))
    + ((qfixed / Bf**2) * (np.cosh(Bf * Lx) - 1) / (np.sinh(Bf * Lx)))
    * np.sinh(Bf * x)
)

critical = lambda x: np.sin((np.pi / Lx) * x)

ytestf = np.zeros_like(xtest)
ytestc = np.zeros_like(xtest)

for i in range(len(xtest)):
    ytestf[i] = fixed(xtest[i])
    ytestc[i] = critical(xtest[i])

ytestc = ytestc * np.linalg.norm(ytestf, np.Inf)

plt.figure()
plt.title("One-Dimension, One-Group, Fixed and Critical")
plt.plot(xtest, ytestc, "-")
plt.plot(xtest, ytestf, "--")
plt.xlabel("x [cm]")
plt.ylabel("$\phi(x)$")
plt.legend(["Criticality", "Fixed Source"])
plt.tight_layout()
plt.savefig("../figs/fixed_critical.pdf")
